### PR TITLE
Fix updates in RepositoryEntityListener

### LINKED ISSUE #2381: RepositoryEntityListener overwrites valid data with null
### Description

When saving an existing entity that already exists in the database, the `RepositoryEntityListener` will overwrite fields not present in the current transaction with null. This is a regression when we added upsert functionality. We used to find then save updatable domain objects in `EntityRecordItemListener`. Since we removed the find in favor of upsert logic, the domain object in `RepositoryEntityListener` will be incomplete.

### Steps to reproduce

1. Switch from sql to repository entity listener by default
2. Run `EntityRecordItemListener*Test`
3. A number of tests fails

### Additional context

_No response_

### Hedera network

other

### Version

v0.38.0-rc1

### Operating system

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:
Fix `RepositoryEntityListener` clearing existing valid data on updates

**Related issue(s)**:
Fixes #2381 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/repository/RepositoryEntityListener.java [hunk 1]
```diff
@@ -131,7 +137,8 @@ public void onLiveHash(LiveHash liveHash) throws ImporterException {
 
     @Override
     public void onNft(Nft nft) throws ImporterException {
-        nftRepository.save(nft);
+        Nft merged = nftRepository.findById(nft.getId()).map(existing -> mergeNft(existing, nft)).orElse(nft);
```
Comment (ijungmann) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/repository/RepositoryEntityListener.java:140]: nit: would prefer the steps be put on separate lines like `onEntity`, etc, just to be consistent and clear.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/repository/RepositoryEntityListener.java [hunk 2]
```diff
@@ -146,22 +153,25 @@ public void onNonFeeTransfer(NonFeeTransfer nonFeeTransfer) throws ImporterExcep
...
+        Schedule merged = scheduleRepository.findById(schedule.getConsensusTimestamp())
+                .map(existing -> mergeSchedule(existing, schedule))
+                .orElse(schedule);
+        scheduleRepository.save(merged);
     }
 
     @Override
     public void onToken(Token token) throws ImporterException {
-        tokenRepository.save(token);
+        Token merged = tokenRepository.findById(token.getTokenId()).map(existing -> mergeToken(existing, token))
```
Comment (ijungmann) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/repository/RepositoryEntityListener.java:164]: nit: separate lines here too.

**Review by ijungmann — CHANGES_REQUESTED** [review]
One nit, otherwise looks good.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/repository/RepositoryEntityListener.java (hunk 2 above)
Comment (steven-sheehy) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/repository/RepositoryEntityListener.java]: Done

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/repository/RepositoryEntityListener.java (hunk 1 above)
Comment (steven-sheehy) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/repository/RepositoryEntityListener.java]: Done

**Review by ijungmann — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**